### PR TITLE
Switch FCM notifications to new Google Firebase v1 API  (Adresses #5921)

### LINKED ISSUE #5921: Googles notification FCM Legacy HTTP protocol gets deprecated after 6/21/2024
image

We're using the FCM Legacy HTTP protocol for sending notifications from domoticz to the mobile apps.
https://firebase.google.com/support/faq#fcm-23-deprecation

Next to that we're using Server keys to authenticate that call, which is going to be decommissioned.
image
image

The actions to update this call looks pretty straight forward btw
https://firebase.google.com/docs/cloud-messaging/migrate-v1

### PR DESCRIPTION
**Pull request by kiddigital**
This PR is to address deprecation of Google legacy FCM and move to the new v1 API of Google's Firebase Cloud Messaging.

It now uses the new v1 API and requests an Access Token each time it sends out messages (unless it can get it from its cache and has not expired).

Info from Google can be found [here](https://firebase.google.com/docs/cloud-messaging/migrate-v1).

### CODE REVIEW

**Comment by kiddigital** [comment]
@galadril , this PR still needs proper testing as I don't have (m)any Android mobile devices available for testing (which also needs the latest (beta) App as well I presume?) Could you (build and) test this domoticz version?

**Comment by kiddigital** [comment]
@gizmocuz , this PR is ready for review but not perfect and not thoroughly tested.

But as the old FCM implementation has been deprecated and (probably) not working anymore, I don't see much harm in merging this into the newest Beta so it can be tested more easily by different users.

Any improvements, bugs, etc. can be fixed on the Dev branch. Agree?

**Comment by kiddigital** [comment]
@gizmocuz , if you are ok with the current code, please merge it so others can test it as well in a new Beta. Thx 😃

File: notifications/NotificationFCM.h [hunk 1]
```diff
@@ -10,4 +10,15 @@ class CNotificationFCM : public CNotificationBase {
       protected:
 	bool SendMessageImplementation(uint64_t Idx, const std::string &Name, const std::string &Subject, const std::string &Text, const std::string &ExtraData, int Priority, const std::string &Sound,
 				       bool bFromNotification) override;
+	  private:
+	std::string GAPI_FCM_ProjectID;
```
Comment (gizmocuz) [inline notifications/NotificationFCM.h]: Could you add m_ before all member variables?

**Comment by rezzalex** [comment]
The current fcm notifications are still working so far. Waiting for the mobile app implementation from @galadril, I can imagine having no notification at all, correct ?

**Review by galadril — APPROVED** [review]

**Review by gizmocuz — APPROVED** [review]

**Comment by galadril** [comment]
Did some more testing and some adjustments are needed:
- we can remove the  whole notification json section
- adjust the data section with the next fields: (more is possible ofcourse)
message, subject, deviceid en body

actually like done before, data wise
`sstr << R"(], "data" : { "subject": ")" << Subject << R"(", "body": ")" << Text << R"(", "extradata": ")" << ExtraData << R"(", "priority": ")" << std::to_string(Priority) << "\", ";`

image

**Comment by kiddigital** [comment]
@galadril , the changes you asked for have been applied.

**Pull request merged**